build/build.gradle

```diff
@@ -112,29 +112,19 @@ void generateP2Repository(String title, String antTarget, String metadataReposit
   delete(file("${bnd_eclipseDirectory}/p2"                                    ))
 }
 
-
-/*
- * Setup tasks (p2 features)
- */
-task p2FeatureTree(type: Copy) {
+def generateFeatureXmlTokens() {
   def masterVersion = String.format('%s.%s-%tY%<tm%<td-%<tH%<tM%<tS-g%.7s',
                        bnd('base-version'),
                        bnd('base-version-qualifier'),
                        new Date(),
                        bnd('Git-SHA'))
-  from    'feature'
-  into    "${buildDir}/features"
-  filter(ReplaceTokens, tokens: ['master-version' : masterVersion, 'base-version' : bnd('base-version')])
-  include '**/category.xml'
-  include '**/feature.xml'
-}
-
-task p2Feature(type: Zip) {
-  dependsOn p2FeatureTree
-  destinationDir = file("${buildDir}/features")
-  archiveName    = 'bndtools.feature.jar'
-  from             "${buildDir}/features/bndtools"
-  include          'feature.xml'
+  def tokens = ['master-version' : masterVersion, 'base-version' : bnd('base-version')]
+  def plugins = fileTree("${buildDir}/plugins").getFiles()
+  plugins.each {
+    Jar jar = new Jar(it)
+    tokens.put(jar.getBsn() + '-version', jar.getVersion())
```

Comment:
You need to close these jars. When gradle runs in daemon mode, it can hold them open for a long time and Windows hates open files...

```groovy
new Jar(it).withCloseable { jar ->
  tokens.put(jar.getBsn() + '-version', jar.getVersion())
}
```